godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/TypeManager.kt

```diff
@@ -0,0 +1,102 @@
+package godot.core
+
+import kotlinx.cinterop.*
+import kotlin.native.concurrent.AtomicReference
+import kotlin.native.concurrent.freeze
+
+// TODO: remove when the class generator is done.
+open class Object(__ignore: String?) {
+    constructor(): this(null) {
+        if (Godot.shouldInitPtr()) {
+            // call godot constructor here
+        }
+    }
+
+    internal lateinit var ptr: COpaquePointer
+
+    fun getClass(): String = ""
+}
+object ClassDB {
+    fun getParentClass(cls: String): String = ""
+}
+
+internal object TypeManager {
+    private val tags = AtomicReference(mutableListOf<COpaquePointer>().freeze())
+
+    /**
+     * Register a user defined type.
+     */
+    fun registerUserType(nativescriptHandle: COpaquePointer, className: String, factory: () -> Object) {
```

Comment:
Do we have to call this now from the entry file?

Reply:
Yes, after we register a class to godot.